### PR TITLE
dssp: 4.0.5 -> 4.2.2.1

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

https://github.com/PDB-REDO/dssp/compare/v4.0.5...v4.2.2.1

- update a version-specific dependency, libcifpp 5.0.8
- add a new dependency, libmcfp 1.2.3

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/dssp/default.nix [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
     repo = pname;
     rev = "v${version}";
-    sha256 = "1x35rdcm4fch66pjbmy73lv0gdb6g9y3v023a66512a6nzsqjsir";
+    sha256 = "sha256-vmGvC5d8LTo+pcY9sxwj0d6JvH8Lyk+QSOZo5raBci4=";
   };
 
   nativeBuildInputs = [ cmake ];
 
-  buildInputs = [ boost libcifpp zlib ];
+  buildInputs = [ libcifpp-508 libmcfp zlib ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/dssp/default.nix:28]:
```suggestion
  buildInputs = [ libcifpp' libmcfp zlib ];
```

File: pkgs/applications/science/biology/dssp/default.nix [hunk 2]
```diff
@@ -1,19 +1,31 @@
-{ lib, stdenv, fetchFromGitHub, boost, cmake, libcifpp, zlib, }:
+{ lib, stdenv, fetchFromGitHub, cmake, libcifpp, libmcfp, zlib, }:
+let
+  libcifpp-508 = libcifpp.overrideAttrs (oldAttrs: {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/dssp/default.nix:3]:
```suggestion
  libcifpp' = libcifpp.overrideAttrs (oldAttrs: {
```

File: pkgs/applications/science/biology/dssp/default.nix [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
+    # dssp 4.2.2.1 requires specific version "5.0.8" of libcifpp
+    version = "5.0.8";
+    src = fetchFromGitHub {
+      owner = oldAttrs.src.owner;
+      repo = oldAttrs.src.repo;
+      rev = "v{version}";
+      sha256 = "sha256-KJGcopGhCWSl+ElG3BPJjBf/kvYJowOHxto6Ci1IMco=";
+    };
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/dssp/default.nix:11]:
```suggestion
    src = oldAttrs.src.override {
      rev = "v{version}";
      sha256 = "sha256-KJGcopGhCWSl+ElG3BPJjBf/kvYJowOHxto6Ci1IMco=";
    };
```

File: pkgs/development/libraries/libcifpp/default.nix [hunk 4]
```diff
@@ -1,19 +1,23 @@
-{ lib, stdenv, fetchFromGitHub, boost, cmake, }:
+{ lib, stdenv, fetchFromGitHub, boost, cmake, zlib, }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libcifpp/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, boost, cmake, zlib }:
```

File: pkgs/development/libraries/libcifpp/default.nix [hunk 5]
```diff
@@ -1,19 +1,23 @@
...
-  version = "4.2.2";
+  version = "5.0.8";
 
   src = fetchFromGitHub {
     owner = "PDB-REDO";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0mhplcpni4p8lavrq4fz9qq8mbxhvpnlxzy55yrz8y07d76ajg6y";
+    sha256 = "sha256-KJGcopGhCWSl+ElG3BPJjBf/kvYJowOHxto6Ci1IMco=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libcifpp/default.nix:12]:
```suggestion
```
there are no submodules

File: pkgs/development/libraries/libmcfp/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{ lib, stdenv, fetchFromGitHub, cmake, }:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libmcfp/default.nix:2]:
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake }:

stdenv.mkDerivation rec {
```

File: pkgs/development/libraries/libcifpp/default.nix (hunk 5 above)
Comment (natsukium) [reply pkgs/development/libraries/libcifpp/default.nix]: I don't remember why this appeared, but thanks for finding it.

**Comment by natsukium** [comment]
Thank you for your reviewing.

**Pull request merged**